core/arch/arm/tee/entry_std.c

```diff
@@ -221,6 +222,12 @@ static void entry_invoke_command(struct thread_smc_args *smc_args,
 	struct tee_dispatch_invoke_command_out out;
 	struct optee_msg_param *params = OPTEE_MSG_GET_PARAMS(arg);
 
+#ifdef CFG_TEE_BENCHMARK
+	if (params[TEE_BENCH_DEF_PARAM].u.tmem.buf_ptr)
+		tee_add_timestamp((void *)(uintptr_t)
+			params[TEE_BENCH_DEF_PARAM].u.tmem.buf_ptr,
+			TEE_BENCH_CORE);
+#endif
```

Comment:
I was trying to keep the same implementation of tee_add_timestamp function within different OP-TEE layers, and in different OP-TEE layers we need to handle params differently.
For example, in **optee linux kmod** drive we need to tee_shm_get_va in advance:
```
optee_add_timestamp(tee_shm_get_va(
							param[OPTEE_BENCH_DEF_PARAM].u.memref.shm,
							0), OPTEE_BENCH_KMOD);
```

in **optee_client** we work with different tee_ioctl_param, then in NW app:
```
struct tee_ioctl_param {
	__u64 attr;
	union {
		struct tee_ioctl_param_memref memref;
		struct tee_ioctl_param_value value;
	} u;
};
```
at the same time in **optee core** params are represented by this structure:
```
struct optee_msg_param {
	uint64_t attr;
	union {
		struct optee_msg_param_tmem tmem;
		struct optee_msg_param_rmem rmem;
		struct optee_msg_param_value value;
	} u;
};
```

I'm afraid that if we move handling of all these cases into **tee_add_benchmark** we will receive a huge "always inlined" monster (or as an option we need to keep different implementations for different layers)